O365 integration (#255)
owncloud/docs-server: modules/admin_manual/pages/enterprise/collaboration/msoffice-wopi-integration.adoc

* Out-of-the box only the on-premise version of Microsoft Office Online Server is supported.
* If you want to integrate the {office365-url}[Office 365 (cloud)] version of Microsoft Office Online, you need to {oc-contact-url}[get in touch with us].
* This app requires at minimum ownCloud Version 10.1 and php 7.1.
====

== Preparing the Environment
You need an {office-online-server-url}[Office Online Server] installed or an active Microsoft 365 subscription.

All involved servers (Office Online Server and the ownCloud server) need to be accessible by HTTPS with valid certificates.

== Configuring the WOPI App in ownCloud

To configure the WOPI app in your ownCloud installation, add the following configuration to `config/config.php`, and adjust it based on the details of your setup:

[source,php]
----
# WOPI token:
# For Office Online Server: Replace the token with your own random string
# For Office 365 (cloud): {oc-contact-url}[Request the string from us] (has to match the `O365_PROXY_SHARED_KEY` configuration of the O365 proxy)
'wopi.token.key' => 'REPLACE_WITH_WOPI_TOKEN_KEY'

# Office server URL

> # For Office Online Server: Enter your your.office.online.server.tld

Reviewer: I have updated the link definition so it will not get a clickable pointing to nowhere

# For Office 365 (cloud):
# Upstream url to Microsoft O365. Microsoft will only accept connections from registered ownCloud domains.
# For Office 365 quality assurance upstream URL: https://ffc-onenote.officeapps.live.com/hosting/discovery
# For Office 365 production upstream URL: https://onenote.officeapps.live.com/hosting/discovery
'wopi.office-online.server' => 'https://the.office.online.server.tld',

# Proxy URL
# Only for Office 365 (cloud), not needed for Office Online Server
# URL of the O365 proxy instance, the path `/wopi/files/` is required.

> 'wopi.proxy.url' => 'https://wopi.owncloud.com/wopi/files/',

Reviewer 2: curl https://wopi.owncloud.com/wopi/files/
curl: (6) Could not resolve host: wopi.owncloud.com

??? should that work?

# Samesite Cookie
# Only for Office 365 (cloud), not needed for Office Online Server
# Necessary to allow e.g. opening ownCloud sharing from O365.

> # Use `None` in case you are using OpenID Connect.

Reviewer 3: ```suggestion
# Use `None` if you are using OpenID Connect.
```
